src/App.tsx

```diff
@@ -15,36 +17,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  String = 'alphabetical',
+  Number = 'length',
+  Default = '',
+}
+
 export const App: React.FC = () => {
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.Default);
+  const [reversed, setReversed] = useState<boolean>(false);
+  const sortedGoods = [...goodsFromServer].sort((good1: string, good2: string) => {
+    switch (sortOrder) {
+      case 'alphabetical':
+        return good1.localeCompare(good2);
+      case 'length':
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    sortedGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortOrder !== 'alphabetical',
+          })}
+          onClick={() => setSortOrder(SortType.String)}
```

Comment:
[CODE KNOWLEDGE] You are directly interacting with the DOM by setting the 'onClick' property. Instead, consider using the 'useCallback' hook from React. This will ensure your function is not recreated on every render, improving performance.